### ISSUE TITLE
Set up common scrape tooling

### ISSUE THREAD

**Issue by reefdog** [TechAndCheck/tech-and-check-alerts#20]
We know we're going to want some common tooling shared across the scrapers, irrespective of site. Create a Hello World scraper that demonstrates:

- [x] Fetching a specific URL
- [x] Navigating the DOM looking for a certain type of link and following it
- [x] Extracting the target contents of the page
- [x] Getting the content in a state ready to be passed to the parsing step (ClaimBuster API, etc.) in an error-recoverable way (e.g., we shouldn't lose the data if the API call doesn't succeed)

**Comment by slifty** [comment]
From what I have read, it looks like [cheerio](https://www.npmjs.com/package/cheerio) is the standard package used for navigating DOM / scraping in Node.

I'm spending some time right now looking at potential architecture for the general tooling; we should keep in mind that the nature of our scrapes make it so the "general" tooling is probably going to be third party libraries like cheerio. For instance, I think the CNN transcript crawl logic is going to be unique to CNN.

Basically, since we're very targeted in our scraping and completion is so important, I think the "abstract" parts of this scrape is really more at a pipeline and error handling level.

**Comment by slifty** [comment]
I am looking to use a queue manager to keep track of crawls and scrapes -- realizing maybe this should be an issue in itself, but I'll just lump it in this anyway.

Looking around it seems like [Kue](https://www.npmjs.com/package/kue) used to be the go to lib for node, but it hasn't been updated in two years so instead I'll be looking at [bull](https://www.npmjs.com/package/bull).  Will circle back if I end up finding something else entirely.

**Comment by slifty** [comment]
I'm not QUITE ready to close this because I would like to create something that more gracefully handles HTTP errors etc.  Maybe that should happen as a separate issue (since for now I think it probably handles it reasonably well actually; in that it would write an error to the log and crash that specific scrape process which is basically what would happen anyway)

@reefdog what do you think?

**Comment by reefdog** [comment]
I say close this and open a new enhancement issue for that.
